rancher/distros-test-framework
# Airgap with Private Registry - part1

#### Proposed Changes ####


- Adding support for airgap testing using private registry method


- Yes, TBD

#### Types of Changes ####


New Functionality
- Added new env var `ENV_MODULE` that accepts terraform module value
- Added new module `/module/airgap`
- Added private registry setup scripts and dependencies
- Added test to create airpgapped cluster
- Validations on nodes, pods, workloads [PENDING]

#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.
Yes, Jenkins vars will be updated

2. Verify code lint; we should not have errors.
- ok

3. Update the documentation if needed.
- TBD

4. Update makefile and docker run if adding new tests.
- ok

5. Run your tests at least 4 times with all configurations needed and possible.
- ok

6. If needed test with different os types.
- ok

#### Linked Issues ####


https://github.com/rancher/qa-tasks/issues/1513

#### Further Comments ####
